dali/python/backend_impl.cc

```diff
@@ -1084,13 +1095,20 @@ PYBIND11_MODULE(backend_impl, m) {
         "show_tensors"_a = false,
         "show_ids"_a = false,
         "use_colors"_a = false)
-    .def("epoch_size", &Pipeline::EpochSize)
-    .def("epoch_size",
+    .def("reader_meta", [](Pipeline* p) {
+          std::map<std::string, ReaderMeta> meta_map = p->GetReaderMeta();
+          py::dict d;
+          for (auto const& value : meta_map) {
+            d[value.first.c_str()] = MetaToDict(value.second);
+          }
+          return d;
+        })
+    .def("reader_meta",
         [](Pipeline* p, const std::string& op_name) {
-          std::map<std::string, Index> sizes = p->EpochSize();
-          DALI_ENFORCE(sizes.find(op_name) != sizes.end(),
-              "Operator " + op_name + " does not expose valid epoch size.");
-          return sizes[op_name];
+          ReaderMeta meta = p->GetReaderMeta(op_name);
+          DALI_ENFORCE(meta,
+              "Operator " + op_name + " does not expose valid metadata.");
```

Comment:
```suggestion
              "Operator " + op_name + " not found or does not expose valid metadata.");
```
because this happens when you misspell the name

Reply:
Done